bookserver/schemas.py

```diff
@@ -35,7 +35,9 @@ class LogItemIncoming(BaseModel):
     to add additional constraints we can do so.
     """
 
-    # :index:`TODO`: Is there any way to specify max string lengths? The database has fixed-length fields for some of these.
+    # FIXME: Use max lengths for strings based on the actual lengths from the database using `Pydantic constraints <https://pydantic-docs.helpmanual.io/usage/types/#constrained-types>`_. Is there any way to query the database for these, instead of manually keeping them in sync?
+    #
+    # Even better, we should introspect the database and auto-convert this to Pydantic. See the idea at https://github.com/tiangolo/pydantic-sqlalchemy/blob/master/pydantic_sqlalchemy/main.py. See also https://github.com/kolypto/py-sa2schema. Other ideas: https://docs.sqlalchemy.org/en/14/core/reflection.html,
```

Comment:
I had looked at that too.  I don't like the idea of redoing model work if we don't have to.  A few months ago the comments were that the pydantic-sqlalchemy was not ready for prime time.  But agree that we should nail down a better solution now rather than later.

Reply:
I posted an issue for this.